packages/story-editor/src/components/library/panes/pageTemplates/renameDialog.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { __ } from '@googleforcreators/i18n';
+import { Input } from '@googleforcreators/design-system';
+import { useState } from '@googleforcreators/react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from '../../../dialog';
+
+const InputWrapper = styled.form`
+  margin: 16px 4px;
+  width: 470px;
+  height: 100px;
+`;
+
+/**
+ * Display a dialog for when a user wants to rename a template.
+ *
+ * @param {Function} props.onClose Callback to toggle dialog display on close.
+ * @param {Function} props.onUpdateName Callback to save template. Template name will be the argument
+ * @param {string} props.previousName Previous name of the template
+ * @return {null|*} The dialog element.
+ */
+
+function RenameDialog({ onClose, onUpdateName, previousName = '' }) {
```

Comment:
`RenameDialog` and `SaveDialog` are very very similar.

Can we just use 1 dialog with different props for both use cases?